### PR TITLE
cmd,interfaces/mount: run snap-update-ns and snap-discard-ns from core if possible

### PR DESCRIPTION
**Pull request by zyga**
This patch fixes "re-exec" of the two aforementioned tools by adding a new helper `cmd.InternalToolPath` and using it in the right spot. I will follow up with a change to use the same
helper for `snap-confine`

### CODE REVIEW

**Comment by zyga** [comment]
@chipaca yes, I want to fix the immediate issue but I'm already writing some tests for that (we had none before)

File: cmd/cmd.go [hunk 1]
```diff
@@ -46,37 +47,119 @@ const newCore = "/snap/core/current"
...
+		return false
+	}
+	return true
+}
+
+// coreSupportsReExec returns true if the given core snap should be used as re-exec target.
+//
+// Ensure we do not use older version of snapd, look for info file and ignore
+// version of core that do not yet have it.
+func coreSupportsReExec(corePath string) bool {
```
Comment (mvo5) [inline cmd/cmd.go]: I would prefer a different name for this helper, most cores "support" re-exec, this code is about if the version of snapd in core is higher than the version currently running. Maybe `newerSnapdIn("/path/to/core")` ?

File: cmd/cmd.go [hunk 2]
```diff
@@ -46,37 +47,119 @@ const newCore = "/snap/core/current"
...
+	// If the distribution doesn't support re-exec or run-from-core then don't do it.
+	if !distroSupportsReExec() {
+		return distroTool
+	}
+
+	// Is the tool we are after present in the core snap?
+	corePath := newCore
+	coreTool := filepath.Join(newCore, "/usr/lib/snapd", tool)
+	if !osutil.FileExists(coreTool) {
+		corePath = oldCore
```
Comment (mvo5) [inline cmd/cmd.go]: Not sure we still need to support oldCore at this point. This code will only be build in the core snap. We have frozen ubuntu-core at version 2.25.

File: interfaces/mount/ns.go [hunk 3]
```diff
@@ -38,7 +39,7 @@ func mountNsPath(snapName string) string {
 func runNamespaceTool(toolName, snapName string) ([]byte, error) {
 	mntFile := mountNsPath(snapName)
 	if osutil.FileExists(mntFile) {
-		toolPath := filepath.Join(dirs.DistroLibExecDir, toolName)
+		toolPath := cmd.InternalToolPath(toolName)
```
Comment (mvo5) [inline interfaces/mount/ns.go]: Feels like a slightly odd place `cmd.InternalToolPath` but then it also fits - it is about commands afterall.

File: cmd/cmd.go (hunk 2 above)
Comment (zyga) [inline cmd/cmd.go]: I wanted to keep it as is but I can remove it. If you agree I'll do it in a separate PR (I've started working on tests already)

File: interfaces/mount/ns.go (hunk 3 above)
Comment (zyga) [inline interfaces/mount/ns.go]: Yes, I was wondering where this all ought to live. Ideas welcome, I'll happily move this elsewhere

**Pull request merged**